### DISCUSSION TITLE
How do I contribute?

### DISCUSSION THREAD

**Discussion by Sushant1708** [microsoft/Qcodes#5267] [Q&A]
Hello, 
I am new to GitHub and haven't used it too much before. I have written a driver for NF Corporation's ZM2376 LCR meter and I wanted to get it added to qcodes.
I forked the repository, created a folder for the maker, added my driver in it and committed the change by submitting a pull request. Then, GitHub led me to merging and down some other complex path, so I don't know if my request was recorded.
As a 2nd question, I have an example file too but it uses the .py driver file from my computer. How do I do it in the format qcodes wants?
Thank you.

**Comment by jenshnielsen** [comment]
Hi @Sushant1708 

Thanks for your question. I think you are on the right track. We normally merge new drivers into https://github.com/qcodes/qcodes_contrib_drivers so that is probably where you want to add it.

The normal github flow is as you describe

* create a fork of the repo with your own user name
* Make a branch in that fork and add the changes that you wish to contribute
* Open a pull request with the changes. 

I have not seen your pull request so I think something went wrong there. If I had to guess I think you may have submitted the pull request against your own fork rather than the official one.

Once you have pushed your changes to a branch of qcodes_contrib_drivers you should be able to go to https://github.com/QCoDeS/Qcodes_contrib_drivers/pulls and click the new pull request button to open a pull request. Hope that helps


With respect to the example. Most of our examples are jupyter notebooks. These are intended to be easy to create and add. I would recommend that you install jupyter lab and copy your example code into a notebook with a suitable name in the examples folder
